### PR TITLE
fix(SankeyChart): skip non-finite node and link geometry

### PR DESCRIPTION
**Pull request by saurav12**
## Description

A SankeyChart node with no links renders invalid SVG. Recharts derives each node's value from its links, so a link-less node has no value to lay out and its `x`, `y` and `height` come back as `NaN`. Those values flowed straight into the rendered output as `<rect x="NaN" height="NaN" y="NaN">`, and React logged `Warning: Received NaN for the \`x\` attribute`.

`renderLink` had the same defect in a quieter form: `NaN` landed inside the `d` path string, so React logged nothing at all and the path was silently malformed.

The existing test only asserted `.not.toThrow()`, so it passed while rendering broken output.

This PR also refreshes the stale SankeyChart snapshot for the inline label layout, which was already failing on `master` (unrelated to this fix — it is what surfaced the warning).

## Changes

- `SankeyChart.web.tsx`
  - Added a module-level `hasFiniteGeometry()` helper.
  - `renderNode` returns an empty `<g />` when `x`, `y`, `width` or `height` is not finite.
  - `renderLink` returns an empty `<path />` when any curve coordinate is not finite.
- `SankeyChart.web.test.tsx` — added `does not emit NaN geometry for a node that has no links`, asserting no `NaN` in `rect` geometry attributes or in `path` `d` strings.
- `SankeyChart.web.test.tsx.snap` — updated to the current inline label layout.
- Added a patch changeset.

## Additional Information

Native is unchanged: `SankeyChart.native.tsx` uses a hand-rolled layout rather than recharts, so it has no equivalent code path.

The new test was verified not to be vacuous — with the `renderNode` guard temporarily removed it fails with the three NaN warnings, and passes with the guard restored.

Verification:

| Check | Result |
| --- | --- |
| `yarn typecheck` (web + native) | pass |
| `eslint` on changed files | no errors |
| `yarn test:react` (full) | 691 suites, 2331 passed, 1345 snapshots |
| `yarn test:react-native` (full) | 601 suites, 1630 passed, 1007 snapshots |

## Component Checklist

- [ ] Update Component Status Page
- [ ] Perform Manual Testing in Other Browsers
- [ ] Add KitchenSink Story
- [x] Add Interaction Tests (if applicable)
- [x] Add changeset

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Review by kamaleshs-bridge4 — DISMISSED** [review]
## ✨ Agentic PR Review ✨

**Status:** Approved ✅

### UI Review

✅ 6 passed

<details>
<summary>Passing checks (6)</summary>

| Check | Screenshot |
|-------|------------|
| ✅ [SankeyChart Default rendering](https://61c19ee8d3d282003ac1d81c-tonoifndir.chromatic.com/?path=/story/components-charts-sankeychart--default-sankey-chart) | <a href="https://raw.githubusercontent.com/razorpay/blade/__ci_artifacts/artifacts/review/PR-3967/ui-critique/22-05-34/components-charts-sankeychart--default-sankey-chart-1.png"></a> |
| ✅ [SankeyChart Custom Node Colors](https://61c19ee8d3d282003ac1d81c-tonoifndir.chromatic.com/?path=/story/components-charts-sankeychart--custom-node-colors-sankey-chart) | <a href="https://raw.githubusercontent.com/razorpay/blade/__ci_artifacts/artifacts/review/PR-3967/ui-critique/22-05-34/components-charts-sankeychart--custom-node-colors-sankey-chart-2.png"></a> |
| ✅ [SankeyChart Single Color](https://61c19ee8d3d282003ac1d81c-tonoifndir.chromatic.com/?path=/story/components-charts-sankeychart--single-color-sankey-chart) | <a href="https://raw.githubusercontent.com/razorpay/blade/__ci_artifacts/artifacts/review/PR-3967/ui-critique/22-05-34/components-charts-sankeychart--single-color-sankey-chart-3.png"></a> |
| ✅ [SankeyChart without Labels](https://61c19ee8d3d282003ac1d81c-tonoifndir.chromatic.com/?path=/story/components-charts-sankeychart--sankey-chart-without-labels) | <a href="https://raw.githubusercontent.com/razorpay/blade/__ci_artifacts/artifacts/review/PR-3967/ui-critique/22-05-34/components-charts-sankeychart--sankey-chart-without-labels-4.png"></a> |
| ✅ [SankeyChart with Plain Text Labels](https://61c19ee8d3d282003ac1d81c-tonoifndir.chromatic.com/?path=/story/components-charts-sankeychart--sankey-chart-with-plain-text-labels) | <a href="https://raw.githubusercontent.com/razorpay/blade/__ci_artifacts/artifacts/review/PR-3967/ui-critique/22-05-34/components-charts-sankeychart--sankey-chart-with-plain-text-labels-5.png"></a> |
| ✅ [SankeyChart master comparison](https://blade.razorpay.com/?path=/story/components-charts-sankeychart--default-sankey-chart) | <a href="https://raw.githubusercontent.com/razorpay/blade/__ci_artifacts/artifacts/review/PR-3967/ui-critique/22-05-34/components-charts-sankeychart--default-sankey-chart-6.png"></a> |

</details>

### Usage

```jsx
import { ChartSankey, ChartSankeyWrapper } from '@razorpay/blade/components';

<ChartSankeyWrapper>
  <ChartSankey
    data={{
      nodes: [{ id: 'x', name: 'Orphan' }],
      links: [],
    }}
  />
</ChartSankeyWrapper>
```

**Review by kamaleshs-bridge4 — APPROVED** [review]
## ✨ Agentic PR Review ✨

**Status:** Approved ✅

### UI Review

✅ 5 passed

<details>
<summary>Passing checks (5)</summary>

| Check | Screenshot |
|-------|------------|
| ✅ [SankeyChart default story](https://61c19ee8d3d282003ac1d81c-tonoifndir.chromatic.com/?path=/story/components-charts-sankeychart--default-sankey-chart) | <a href="https://raw.githubusercontent.com/razorpay/blade/__ci_artifacts/artifacts/review/PR-3967/ui-critique/22-34-24/components-charts-sankeychart--default-sankey-chart-1.png"></a> |
| ✅ [SankeyChart custom node colors story](https://61c19ee8d3d282003ac1d81c-tonoifndir.chromatic.com/?path=/story/components-charts-sankeychart--custom-node-colors-sankey-chart) | <a href="https://raw.githubusercontent.com/razorpay/blade/__ci_artifacts/artifacts/review/PR-3967/ui-critique/22-34-24/components-charts-sankeychart--custom-node-colors-sankey-chart-2.png"></a> |
| ✅ [SankeyChart single color story](https://61c19ee8d3d282003ac1d81c-tonoifndir.chromatic.com/?path=/story/components-charts-sankeychart--single-color-sankey-chart) | <a href="https://raw.githubusercontent.com/razorpay/blade/__ci_artifacts/artifacts/review/PR-3967/ui-critique/22-34-24/components-charts-sankeychart--single-color-sankey-chart-3.png"></a> |
| ✅ [SankeyChart without labels story](https://61c19ee8d3d282003ac1d81c-tonoifndir.chromatic.com/?path=/story/components-charts-sankeychart--sankey-chart-without-labels) | <a href="https://raw.githubusercontent.com/razorpay/blade/__ci_artifacts/artifacts/review/PR-3967/ui-critique/22-34-24/components-charts-sankeychart--sankey-chart-without-labels-4.png"></a> |
| ✅ [SankeyChart with plain text labels story](https://61c19ee8d3d282003ac1d81c-tonoifndir.chromatic.com/?path=/story/components-charts-sankeychart--sankey-chart-with-plain-text-labels) | <a href="https://raw.githubusercontent.com/razorpay/blade/__ci_artifacts/artifacts/review/PR-3967/ui-critique/22-34-24/components-charts-sankeychart--sankey-chart-with-plain-text-labels-5.png"></a> |

</details>

### Usage

```jsx
import { SankeyChart } from '@razorpay/blade/components';

<SankeyChart data={{ nodes: [{ id: 'x', name: 'Orphan' }], links: [] }} />
```

File: packages/blade/src/components/Charts/SankeyChart/__tests__/SankeyChart.web.test.tsx [hunk 1]
```diff
@@ -310,6 +310,30 @@ describe('SankeyChart — edge cases', () => {
...
+            nodes: [{ id: 'x', name: 'Orphan' }],
+            links: [],
+          }}
+        />
+      </ChartSankeyWrapper>,
+    );
+
+    // A node with no links has no value for Recharts to lay out, so it must be
+    // skipped rather than rendered with NaN x/y/height attributes.
+    container.querySelectorAll('rect').forEach((rect) => {
```
Comment (kamaleshs-bridge4) [inline packages/blade/src/components/Charts/SankeyChart/__tests__/SankeyChart.web.test.tsx:327]: 🔵 **[MINOR]** · _code-quality-critique_ · confidence: 7/10

**Problem:** The test's `forEach` assertions are likely no-ops with the fix in place. An orphan node with no links causes `renderNode` to return an empty `<g />` (line 539) and `renderLink` is never called (no links), so the container has 0 `<rect>` and 0 `<path>` elements. The `forEach` callbacks never execute, meaning the test passes vacuously without asserting anything. Additionally, the test uses `container.querySelectorAll('rect')` and `container.querySelectorAll('path')` without the `svg` prefix, inconsistent with every other test in the file which uses `svg`-prefixed selectors.

**Suggestion:** Add `expect(container.querySelector('svg')).toBeTruthy()` to verify the component rendered successfully, and assert the expected rect count explicitly (e.g., `expect(container.querySelectorAll('svg rect')).toHaveLength(0)`) so the test's expectations are self-documenting. Use `svg`-prefixed selectors for consistency with the rest of the file.

**Review by kamaleshs-bridge4 — COMMENTED** [review]
## ✨ Agentic PR Review ✨

### UI Review

✅ 5 passed

<details>
<summary>Passing checks (5)</summary>

| Check | Screenshot |
|-------|------------|
| ✅ [Default Sankey Chart](https://61c19ee8d3d282003ac1d81c-ceukgxtgqa.chromatic.com/?path=/story/components-charts-sankeychart--default-sankey-chart) | <a href="https://raw.githubusercontent.com/razorpay/blade/__ci_artifacts/artifacts/review/PR-3967/ui-critique/22-59-48/components-charts-sankeychart--default-sankey-chart-1.png"></a> |
| ✅ [Custom Node Colors Sankey Chart](https://61c19ee8d3d282003ac1d81c-ceukgxtgqa.chromatic.com/?path=/story/components-charts-sankeychart--custom-node-colors-sankey-chart) | <a href="https://raw.githubusercontent.com/razorpay/blade/__ci_artifacts/artifacts/review/PR-3967/ui-critique/22-59-48/components-charts-sankeychart--custom-node-colors-sankey-chart-2.png"></a> |
| ✅ [Single Color Sankey Chart](https://61c19ee8d3d282003ac1d81c-ceukgxtgqa.chromatic.com/?path=/story/components-charts-sankeychart--single-color-sankey-chart) | <a href="https://raw.githubusercontent.com/razorpay/blade/__ci_artifacts/artifacts/review/PR-3967/ui-critique/22-59-48/components-charts-sankeychart--single-color-sankey-chart-3.png"></a> |
| ✅ [Sankey Chart without Labels](https://61c19ee8d3d282003ac1d81c-ceukgxtgqa.chromatic.com/?path=/story/components-charts-sankeychart--sankey-chart-without-labels) | <a href="https://raw.githubusercontent.com/razorpay/blade/__ci_artifacts/artifacts/review/PR-3967/ui-critique/22-59-48/components-charts-sankeychart--sankey-chart-without-labels-4.png"></a> |
| ✅ [Sankey Chart with Plain Text Labels](https://61c19ee8d3d282003ac1d81c-ceukgxtgqa.chromatic.com/?path=/story/components-charts-sankeychart--sankey-chart-with-plain-text-labels) | <a href="https://raw.githubusercontent.com/razorpay/blade/__ci_artifacts/artifacts/review/PR-3967/ui-critique/22-59-48/components-charts-sankeychart--sankey-chart-with-plain-text-labels-5.png"></a> |

</details>

### Usage

```jsx
import { ChartSankeyWrapper, ChartSankey } from '@razorpay/blade/components';

<Box height="420px">
  <ChartSankeyWrapper showTooltip>
    <ChartSankey data={{ nodes, links }} labelUnit="txn" />
  </ChartSankeyWrapper>
</Box>
```

**Review by swapnil-kr1 — APPROVED** [review]
